src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,249 @@
+package duke;
+
+import duke.exception.*;
+import duke.tasks.*;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDate;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Represents the chatbot
+ */
+public class Duke {
+    private static String FILEPATH = "./data/duke.txt";
+    private TaskList list;
+    private Ui ui;
+    private Storage storage;
+    private Parser parser;
+
+    /**
+     * Initializes a Duke object
+     */
+    public Duke() {
+        this.ui = new Ui();
+        this.storage = new Storage(FILEPATH);
+        this.list = new TaskList(this.storage.loadData());
+        this.parser = new Parser();
+    }
+
+    private enum DukeCommand {
+        BYE, LIST,
+        MARK, UNMARK,
+        TODO, DEADLINE, EVENT,
+        DELETE,
+        LISTDATE, FIND
+    }
+
+
+    private void inputResponse() {
+        while (true) {
+            String input;
+            input = this.ui.getUserInput();
+            String[] userInput = this.parser.parseUserInput(input);
+            String dukeQuery = userInput[0];
+            DukeCommand dukeCommand = DukeCommand.valueOf(dukeQuery.toUpperCase());
+
+            try {
+                switch (dukeCommand) {
+                    case BYE :
+                        this.storage.storeData(this.list);
+                        this.ui.exitDisplay();
+                        this.ui.closeInput();
+                        return;
+                    case LIST :
+                        this.ui.displayList(this.list);
+                        break;
+                    case MARK :
+                        this.markComplete(userInput);
+                        break;
+                    case UNMARK :
+                        this.markInComplete(userInput);
+                        break;
+                    case TODO :
+                        this.addTodo(userInput);
+                        break;
+                    case DEADLINE :
+                        this.addDeadline(userInput);
+                        break;
+                    case EVENT :
+                        this.addEvent(userInput);
+                        break;
+                    case DELETE :
+                        this.deleteTask(userInput);
+                        break;
+                    case LISTDATE:
+                        this.displayTasksWithDates(userInput);
+                        break;
+                    case FIND:
+                        this.findTasks(userInput);
+                        break;
+                    default :
+                        throw new DukeInvalidCommandException();
```

Comment:
Nice you of enums, just minor indentation fix where switch and case should be align.